packages/arb-validator/structures/inbox.go

```diff
@@ -204,13 +205,23 @@ func (ms *MessageStack) MarshalForCheckpoint(ctx *ckptcontext.CheckpointContext)
 func (x *InboxBuf) UnmarshalFromCheckpoint(ctx ckptcontext.RestoreContext) (*MessageStack, error) {
 	ret := NewMessageStack()
 	ret.hashOfRest = x.HashOfRest.Unmarshal()
+
+	valueCache, err := cmachine.NewValueCache()
+	if err != nil {
+		return nil, err
+	}
+
 	for i := len(x.Items) - 1; i >= 0; i = i - 1 {
-		val := ctx.GetValue(x.Items[i].Unmarshal())
+		val := ctx.GetValue(x.Items[i].Unmarshal(), valueCache)
 		msg, err := inbox.NewInboxMessageFromValue(val)
 		if err != nil {
 			return nil, err
 		}
-		ret.DeliverMessage(msg)
+		err = ret.DeliverMessage(msg)
+		if err != nil {
+			// ##### Were these errors ignored on purpose? #####
+			continue
+		}
```

Comment:
If an error occurs here, we should just return the error

Also I prefer the form
```go
if err := ret.DeliverMessage(msg); err != nil {
    return nil, err
}
```
for conciseness